test/singa/test_image_transformer.cc

```diff
@@ -0,0 +1,119 @@
+/************************************************************
+*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*
+*************************************************************/
+
+#include "singa/io/transformer.h"
+#include "gtest/gtest.h"
+#include <time.h>
+#include <iostream>
+
+// decide whether to use opencv
+// #include "singa/singa_config.h"
+
+#ifdef USE_OPENCV
+#include <opencv2/highhui/highhui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#endif
+
+using singa::Shape;
+TEST(ImageTransformer, Setup) {
+  singa::ImageTransformer img_transformer;
+
+  singa::TransformerConf conf;
+  conf.set_resize_height(256);
+  conf.set_resize_width(256);
+  conf.set_horizontal_mirror(true);
+  conf.set_image_dim_order("NHWC");
+  conf.add_crop_shape(224u);
+  conf.add_crop_shape(200u);
+
+  img_transformer.Setup(conf);
+  EXPECT_EQ(256, img_transformer.Resize_height());
+  EXPECT_EQ(256, img_transformer.Resize_width());
+  EXPECT_EQ(true, img_transformer.Horizontal_mirror());
+  EXPECT_EQ("NHWC", img_transformer.Image_dim_order());
+  EXPECT_EQ(224u, img_transformer.Crop_shape()[0]);
+  EXPECT_EQ(200u, img_transformer.Crop_shape()[1]);
+}
+
+TEST(ImageTransformer, Apply) {
```

Comment:
randomness exists also in crop in kTrain phase.
so, can only test resize in kTrain